### ISSUE TITLE
Why is branching necessary ?

### ISSUE THREAD

**Issue by SkybuckFlying** [Potherca/git-split-file#3]
I am trying to find a solution for my problem as described here:

https://stackoverflow.com/questions/51114596/re-structure-code-split-into-multiple-files-keep-splitted-files-updated-from-o

Your script seems promising to some degree I have some questions about it though, (haven't tried the script yet)

1. Why is the branching necessary ? Is this a functionality to allow safe merging in case of conflicts ? What kind of conflicts could occur ? Maybe split files already exist ?

2. Is branching perhaps done because git mv can only be done once per branch ? If so great solution, but the "how it works" doesn't really explain it like that ? 

Also once the files are split, and suppose somebody updates the original file, will the changes to the original files be re-applied to the split files by GIT ??? I need a functionality like that.

**Comment by Potherca** [comment]
The question you link to seems to have been deleted but I will try to answer your questions here.

1. The need for branching comes from the way git works. Git can't really "understand" if a file is renamed. It just keeps track of code changes (add/edit/delete). If a file is split into multiple separate files in one go, git looses track. Creating a single branch _per file_ is a sort of hack to get around this limitation.

2. "git mv can only be done once per branch" is more-or-less correct. I am sorry if the documentation is lacking in this respect. This project still needs work (both in the docs _and_ in the code). I've been hoping to put more time in to get to a 100% stable version 1.0.0

Ah, I guess that was the problem that was originally described in the StackOVerflow question. 

In short: No. This script does (currently) not support being "re-run", as it were.

I do like the use-case you suggest though. It could be rather useful in several scenario's I come across in my day-to-day work.

If I manage to allocate time to spend on this project, I might be tempted to implement such a feature.

Does this answer your questions sufficiently?
